packages/rulesets/src/spectral/functions/patch-properties-in-normal-resource-definition.ts

```diff
@@ -0,0 +1,58 @@
+import { getProperties } from "./utils"
+
+export const PatchPropertiesInNormalResourceDefinition = (pathItem: any, _opts: any, ctx: any) => {
+  if (pathItem === null || typeof pathItem !== "object") {
+    return []
+  }
+
+  const path = ctx.path
+  const errors: any = []
+
+  if (!pathItem["put"]) {
+    errors.push({
+      message: "Patch request body MUST contain atleast one or more properties present in the normal resource definition(PUT operation).",
+      path: path,
+    })
+    return errors
+  }
+
+  const patchProperties = []
+  const putProperties = []
+  const patchResponses = pathItem["patch"].responses
+  const putResponses = pathItem["put"].responses
+
+  for (const res in patchResponses) {
+    const value = patchResponses[res]
+    const properties = getProperties(value.schema)
+    if (Object.entries(properties).length > 0) {
+      patchProperties.push(properties)
+    }
+  }
+
+  if (patchProperties.length === 0) {
+    return []
+  }
+
+  for (const res in putResponses) {
+    const value = putResponses[res]
+    const properties = getProperties(value.schema)
+    if (Object.entries(properties).length > 0) {
+      putProperties.push(properties)
+    }
+  }
+
+  for (const patchProp of patchProperties) {
+    for (const putProp of putProperties) {
+      if (JSON.stringify(putProp) === JSON.stringify(patchProp)) {
```

Comment:
It seems too strict for the patch operation

Reply:
I mean the description/title may change and it will pass if the patch contains disallowed property 'type','name',‘id’

Reply:
@jianyexi Shouldn't that(PATCH having disallowed properties) be covered in an other rule?
This rule will only check if PATCH has atleast one property defined in PUT